// @pulumi/cloudflare/worker.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Worker = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/worker:Worker example '<account_id>/<worker_id>'
 * ```
 */
class Worker extends pulumi.CustomResource {
    /**
     * Get an existing Worker resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Worker(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Worker.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Worker.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state ? state.accountId : undefined;
            resourceInputs["createdOn"] = state ? state.createdOn : undefined;
            resourceInputs["logpush"] = state ? state.logpush : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["observability"] = state ? state.observability : undefined;
            resourceInputs["subdomain"] = state ? state.subdomain : undefined;
            resourceInputs["tags"] = state ? state.tags : undefined;
            resourceInputs["tailConsumers"] = state ? state.tailConsumers : undefined;
            resourceInputs["updatedOn"] = state ? state.updatedOn : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accountId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args ? args.accountId : undefined;
            resourceInputs["logpush"] = args ? args.logpush : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["observability"] = args ? args.observability : undefined;
            resourceInputs["subdomain"] = args ? args.subdomain : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tailConsumers"] = args ? args.tailConsumers : undefined;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["updatedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Worker.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Worker = Worker;
/** @internal */
Worker.__pulumiType = 'cloudflare:index/worker:Worker';
